### ISSUE TITLE
StackExchange.Redis.IDatabase mock requires more setup now

### ISSUE THREAD

**Issue by dmytrostruk** [redis/NRedisStack#202]
**NRedisStack Version**: 0.10.0 or higher

**Description**: 
Hi!
As soon as I update version of `NRedisStack` to `0.10.0` or higher, my unit tests with `IDatabase` mock are failing. 

Previously, it was possible to mock `IDatabase` easier with: 
`mockDatabase.Setup(x => x.ExecuteAsync(...)).ReturnsAsync(...);`

Now, it requires to mock not only `ExecuteAsync` method, but also:
- `IDatabase.Multiplexer`
- `IConnectionMultiplexer.GetServer(...)`
- `IConnectionMultiplexer.GetEndPoints()`
- `IServer.Version`. 

I believe this is because of this line:
https://github.com/redis/NRedisStack/blob/954992e425934318cda309a27456db8d16e3cdb4/src/NRedisStack/Auxiliary.cs#L75

It was added in this PR: 
https://github.com/redis/NRedisStack/pull/180/files#diff-325a736b58343c97ea736c8f94320677e56eb9eaa2f6b8152ca732e41f8fd197R76

If this is desired implementation, maybe it worth to mention this change as breaking one, so users will know that additional mock setup needs to be added, so it will be possible to test classes that have `NRedisStack` as dependency. 

Thank you!

**Comment by shacharPash** [comment]
Hi @dmytrostruk , thanks for the detailed issue!
I am checking if it is possible to solve it somehow so that you can do mocking as before without additional setup.
I will update.

**Comment by shacharPash** [comment]
@dmytrostruk Can you send me a simple example of a failed test? I want to see if I can fix it

**Comment by dmytrostruk** [comment]
Hi @shacharPash , sorry for late response, I didn't have a chance to check it until now. 

I use `IDatabase` to create index and perform manipulations with data. In my unit tests I need to properly mock `IDatabase`, so I don't perform actual requests.

To create index, I use this code:
```csharp
var ft = database.FT();
await ft.CreateAsync(...);
```

Taking into account, that `IDatabase.FT()` returns type `SearchCommands` instead of abstraction, I cannot mock it in my tests, so I have to mock underlying `SearchCommands` logic.

So, here is test example to create index with required mocks, that works using NRedisStack v0.9.0:
```csharp
[Fact]
public async Task CreateAsync()
{
    // Arrange
    const string Collection = "test_collection";

    var mockDatabase = new Mock<IDatabase>();
    mockDatabase
        .Setup<Task<RedisResult>>(x => x.ExecuteAsync(
            It.Is<string>(x => x == "FT.CREATE"),
            It.Is<object[]>(x => x[0].ToString() == Collection))
        )
        .ReturnsAsync(RedisResult.Create("OK", ResultType.SimpleString));

    var ft = mockDatabase.Object.FT();

    var createParams = FTCreateParams.CreateParams();
    var schema = new Schema().AddTextField("key");

    // Act
    var result = await ft.CreateAsync(Collection, createParams, schema);

    // Assert
    Assert.True(result);
}
```

As soon as I update to v0.11.0, it throws following exception:
![image](https://github.com/redis/NRedisStack/assets/13853051/6ff1c388-1ee6-4a6f-9852-7a3434f595aa)

So, in order to fix this error, I need to update the test with additional mocks, as I shared in issue description above. Here is updated test:
```csharp
    [Fact]
    public async Task CreateAsync()
    {
        // Arrange
        const string Collection = "test_collection";

        var mockDatabase = new Mock<IDatabase>();
        mockDatabase
            .Setup<Task<RedisResult>>(x => x.ExecuteAsync(
                It.Is<string>(x => x == "FT.CREATE"),
                It.Is<object[]>(x => x[0].ToString() == Collection))
            )
            .ReturnsAsync(RedisResult.Create("OK", ResultType.SimpleString));

        // Additional setup to fix test for NRedisStack v0.11.0
        var mockBatch = new Mock<IBatch>();

        mockDatabase.Setup(x => x.CreateBatch(It.IsAny<object>())).Returns(mockBatch.Object);

        var mockConnectionMultiplexer = new Mock<IConnectionMultiplexer>();
        var mockServer = new Mock<IServer>();
        var mockEndpoint = new Mock<EndPoint>();

        mockServer.Setup(x => x.Version).Returns(new Version());

        mockConnectionMultiplexer.Setup(x => x.GetServer(It.IsAny<string>(), It.IsAny<object>())).Returns(mockServer.Object);
        mockConnectionMultiplexer.Setup(x => x.GetEndPoints(It.IsAny<bool>())).Returns(new EndPoint[] { mockEndpoint.Object });

        var ft = mockDatabase.Object.FT();

        var createParams = FTCreateParams.CreateParams();
        var schema = new Schema().AddTextField("key");

        // Act
        var result = await ft.CreateAsync(Collection, createParams, schema);

        // Assert
        Assert.True(result);
    }
```

As you can see, with new v0.11.0 I need a lot more mocking code to avoid any errors. Maybe this can be simplified to the state how it was in v0.9.0. 

But ideally, I think `IDatabase.FT()` could return some abstraction instead of actual type (if that is possible), so it will be very easy to mock it in tests. 

Please let me know if that makes sense and if I'm missing something. Thanks a lot!

**Comment by dmytrostruk** [comment]
Hi @shacharPash , sorry for late response, I didn't have a chance to check it until now.

It appeared that with new NRedisStack version 0.11.0, the only additional setup to mock `IDatabase` for index creation is the following:
```csharp
var mockBatch = new Mock<IBatch>();
mockDatabase.Setup(x => x.CreateBatch(It.IsAny<object>())).Returns(mockBatch.Object);
```

In this case I believe this issue can be closed now. Thank you!
